src/services/codeFixProvider.ts

```diff
@@ -1,40 +1,56 @@
 /* @internal */
 namespace ts {
-    export interface CodeFix {
+    export interface CodeFixRegistration {
         errorCodes: number[];
-        getCodeActions(context: CodeFixContext): CodeAction[] | undefined;
+        getCodeActions(context: CodeFixContext): CodeFix[] | undefined;
+        groupIds: string[];
+        fixAllInGroup(context: CodeFixAllContext): CodeActionAll;

```

Comment:
this should be optional. exsiting users like tslint for instance do not supply this function.

Reply:
though this si not a public API.

Reply:
Also a name that still include `codeActions` would be better.. something that maintains parity with `getCodeActions`

Reply:
maybe `getAllCodeActions` or `getAllCodeActionsInFile`